### PR TITLE
fix admin notice for beginning i18n and start

### PR DESCRIPTION
**Pull request by afragen**
foundation for plugin name setter.

### CODE REVIEW

**Comment by afragen** [comment]
I've left out the text domain. I think we may just have to put a dummy `my-plugin-slug` and remind the user to change it.

OK, I thought better of it and used a text domain that will need changing.

**Comment by coenjacobs** [comment]
> I think we may just have to put a dummy my-plugin-slug and remind the user to change it.

That's not really going to work since this is a dependency class and should ideally be used as is. When you add it as a dependency to your project via Composer, you don't want to need to edit it. I doubt there is a valid way to do this in a way that supports the WordPress core way of translating things... :cry:

**Comment by afragen** [comment]
I'll pull out the text domain slug. I see what you mean.

**Comment by afragen** [comment]
How's the rest of it?

**Comment by afragen** [comment]
@coenjacobs when I update POT from sources using POEdit it pulls in this string without having an attached text domain :smile:

File: src/WPUpdatePhp.php [hunk 1]
```diff
@@ -47,9 +82,17 @@ private function load_minimum_required_version_notice() {
...
+	 * Generate admin notice
+	 */
 	public function admin_notice() {
-		echo '<div class="error">';
-		echo '<p>Unfortunately, this plugin can not run on PHP versions older than '. $this->minimum_version .'. Read more information about <a href="http://www.wpupdatephp.com/update/">how you can update</a>.</p>';
-		echo '</div>';
+		?>
+		<div class="error">
+			<p>
+				<?php printf( __( 'Unfortunately, %1$s can not run on PHP versions older than %2$s. Read more information about %3$sshow you can update%4$s.' ), $this->plugin_name, $this->minimum_version, '<a href="http://www.wpupdatephp.com/update/">', '</a>' ); ?>
```
Comment (cfoellmann) [inline src/WPUpdatePhp.php]: I would make the url translatable, too.
Translator can set something like `http://www.wpupdatephp.com/de/update/` or whatever the syntax for different languages for the site will be.

**Comment by afragen** [comment]
Let me know if there are other corrections.

**Comment by coenjacobs** [comment]
I still have a couple issues with this approach and this PR becomes a little confusing as well now that there are effectively three changes mixed up again. :smile:

We still don't have a real way to translate strings inside this library, as we can't use a different textdomain as the bundling plugin. More discussion about that in #8.

The changes to the `does_my_plugin_meet_required_php_version` method change its accepted parameters and that's gonna break backwards compatibility. I don't want to do that, plus I think it's overkill to pass the entire `__FILE__` just to get the plugin name. Just passing the name as a string to a set method would do just fine in my opinion.

I'll have a look to split these commits into different branches so we can look at them case by case.

**Comment by coenjacobs** [comment]
I've extracted the plugin name part to #20 and will merge that into the release branch when I'm done.

**Pull request closed without merging**